File: spring-boot-ecommerce/src/main/java/com/fullstack/ecommerce/dto/Purchase.java
package com.fullstack.ecommerce.dto;

import com.fullstack.ecommerce.entity.Address;
import com.fullstack.ecommerce.entity.OrderItem;
import com.fullstack.ecommerce.entity.Customer;
import com.fullstack.ecommerce.entity.Order;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: angular-ecommerce/src/app/models/purchase.ts
import { Address } from './address';
import { Customer } from './cutomer';
import { Order } from './order';
import { OrderItem } from './order-item';

export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}
